server/src/main/java/org/apache/cassandra/sidecar/db/ConfigAccessorImpl.java

```diff
@@ -0,0 +1,108 @@
+import org.apache.cassandra.sidecar.db.schema.ConfigsSchema;
+import org.apache.cassandra.sidecar.db.schema.SidecarSchema;
+import org.apache.cassandra.sidecar.routes.cdc.Service;
+
+/**
+ * Configurations for CDC feature are stored inside a table "config" in an internal sidecar keyspace.
+ * {@link ConfigAccessorImpl} is an accessor for the above-mentioned table and encapsulates database
+ * access operations of the "config" table.
+ */
+public abstract class ConfigAccessorImpl extends DatabaseAccessor<ConfigsSchema> implements ConfigAccessor
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigAccessorImpl.class);
+    private final Service service = service();
+    private final SidecarSchema sidecarSchema;
+
+    protected ConfigAccessorImpl(ConfigsSchema configsSchema,
+                                 CQLSessionProvider sessionProvider,
+                                 SidecarSchema sidecarSchema)
+    {
+        super(configsSchema, sessionProvider);
+        this.sidecarSchema = sidecarSchema;
+    }
+
+    public abstract Service service();
+
+    @Override
+    public ServiceConfig getConfig()
+    {
+        sidecarSchema.ensureInitialized();
+        BoundStatement statement = tableSchema.selectConfig()
+                                              .bind(service.serviceName);
+        Row row = execute(statement).one();
+        if (row == null || row.isNull(0))
+        {
+            LOGGER.debug(String.format("No %s configs are present in the table C* table", service.serviceName));
+            return new ServiceConfig(Map.of());
+        }
+        return ServiceConfig.from(row);
+    }
+
+    @Override
+    public ServiceConfig storeConfig(Map<String, String> config)
+    {
+        sidecarSchema.ensureInitialized();
+        BoundStatement statement = tableSchema.insertConfig()
+                                              .bind(service.serviceName, config);
+        execute(statement);
+        return new ServiceConfig(config);
+    }
+
+    @Override
+    public Optional<ServiceConfig> storeConfigIfNotExists(Map<String, String> config)
+    {
+        sidecarSchema.ensureInitialized();
+        BoundStatement statement = tableSchema.insertConfigIfNotExists()
+                                              .bind(service.serviceName, config);
+        ResultSet resultSet = execute(statement);
+        if (resultSet.wasApplied())
+        {
+            return Optional.of(new ServiceConfig(config));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public void deleteConfig()
+    {
+        sidecarSchema.ensureInitialized();
+        BoundStatement deleteStatement = tableSchema.deleteConfig()
+                                                    .bind(service.serviceName);
+        execute(deleteStatement);
+    }
+
+    @Override
+    public boolean isSchemaInitialized()
+    {
+        return sidecarSchema.isInitialized();
+    }
```

Comment:
override isAvailable instead
```suggestion
        @Override
    public boolean isAvailable()
    {
        return super.isAvailable() && sidecarSchema.isInitialized();
    }
```